crates/control/src/motion/joint_command_optimizer.rs

```diff
@@ -0,0 +1,82 @@
+use color_eyre::Result;
+use context_attribute::context;
+use framework::MainOutput;
+use types::{collected_commands::CollectedCommands, Joints, SensorData};
+
+pub struct JointCommandOptimizer {}
+
+#[context]
+pub struct CreationContext {}
+
+#[context]
+pub struct CycleContext {
+    pub collected_commands: Input<CollectedCommands, "collected_commands">,
+    pub sensor_data: Input<SensorData, "sensor_data">,
+}
+
+#[context]
+#[derive(Default)]
+pub struct MainOutputs {
+    pub optimized_commands: MainOutput<CollectedCommands>,
+}
+
+impl JointCommandOptimizer {
+    pub fn new(_context: CreationContext) -> Result<Self> {
+        Ok(Self {})
+    }
+
+    pub fn cycle(&mut self, context: CycleContext) -> Result<MainOutputs> {
+        let currents = context.sensor_data.current;
+        let collected_commands = context.collected_commands;
+
+        let maximal_current = collected_commands
+            .positions
+            .as_vec()
+            .into_iter()
+            .flatten()
+            .collect::<Vec<f32>>()
+            .into_iter()
+            .fold(0.0, f32::max);
+
+        let optimized_position_angles: [f32; 26] = collected_commands
+            .positions
```

Comment:
Maybe it would make sense to `.iter_mut()` and modify them directly? But no hard feelings. Your functional approach is also good.

Reply:
I like it the way it is :)